QModManager/SummaryLogger.cs

```diff
@@ -0,0 +1,70 @@
+            LogStatus(mods, ModStatus.MissingPatchMethod, "The following mods had no patch methods to run", Logger.Level.Warn);
+            LogStatus(mods, ModStatus.CircularLoadOrder, "The following mods could not be loaded due to circular load order", Logger.Level.Warn);
+            LogStatus(mods, ModStatus.CircularDependency, "The following mods could not be loaded due to circular dependencies", Logger.Level.Warn);
+            LogStatus(mods, ModStatus.MissingDependency, "The following mods could not be loaded due to missing dependencies", Logger.Level.Warn);
+            LogStatus(mods, ModStatus.CurrentGameNotSupported, "The following mods do not support the current game", Logger.Level.Warn);
+            LogStatus(mods, ModStatus.FailedIdentifyingGame, "Could not identify the supported game for the following mods did not ", Logger.Level.Warn);
+            LogStatus(mods, ModStatus.DuplicateIdDetected, "Found the following duplicate mods", Logger.Level.Warn);
+            LogStatus(mods, ModStatus.DuplicatePatchAttemptDetected, "Found the following mods attempted duplicate patching", Logger.Level.Error);
+            LogStatus(mods, ModStatus.MissingCoreInfo, "The following mods could not be loaded for patching due to missing core data", Logger.Level.Warn);
+            LogStatus(mods, ModStatus.InvalidCoreInfo, "The following mods could not be loaded for patching due to invalid core data", Logger.Level.Warn);
+            LogStatus(mods, ModStatus.MissingAssemblyFile, "The following mods had no DLL file to load", Logger.Level.Warn);
+            LogStatus(mods, ModStatus.FailedLoadingAssemblyFile, "The following mods failing loading their DLL files", Logger.Level.Warn);
+            LogStatus(mods, ModStatus.UnidentifiedMod, "The following mods could not be identified for loading", Logger.Level.Warn);
+        }
+
+        private static void LogStatus(PairedList<QMod, ModStatus> mods, ModStatus statusToReport, string summary, Logger.Level logLevel)
+        {
+            if (mods.ValueCount(statusToReport) == 0)
+                return;
+
+            Logger.Log(logLevel, summary);
+            foreach (Pair<QMod, ModStatus> pair in mods)
+            {
+                if (pair.Value == statusToReport)
+                {
+                    QMod mod = pair.Key;
+                    Console.WriteLine($"- {mod.DisplayName} ({mod.Id})");
+                }
+            }
+        }
+
+        private static void CheckOldHarmony(IEnumerable<QMod> mods)
+        {
+            var modsThatUseOldHarmony = new List<QMod>();
+
+            foreach (QMod mod in mods)
+            {
+                if (mod.IsLoaded && mod.HarmonyOutdated)
+                    modsThatUseOldHarmony.Add(mod);
+            }
+
+            if (modsThatUseOldHarmony.Count > 0)
+            {
+                Logger.Warn($"Some mods are using an old version of harmony! This will NOT cause any problems, but it's not recommended:");
```

Comment:
Move this to `LogSummaries`

Reply:
This method is already called from the `LogSummaries` method.  
Did you have something else in mind?

Reply:
So far all of the calls to `LogStatus` are for errored/cancelled mods, that is not where _warn_ logging is done.

Reply:
True. But I don't see any better place to put this, unless it's moved somewhere else entirely.